src/gas-api.js

```diff
@@ -1,210 +1,212 @@
-        })
-        .then((response) => {
-          if (!response.startsWith('//OK'))  reject(new Error('get apps script code failed'));
-          //evil eval, but it's simple to get the object since it's not valid json object
-          const codeContent = eval(response.slice(4)).filter((e) => {
-            return typeof(e) === 'object';
-          })[0];
-          resolve({file : codeContent[codeContent.length - 6], content: codeContent[codeContent.length - 10], id : id });
-        })
-        .fail(reject);
-      })
+    .catch((err) => {
+      showAlert(err.message, LEVEL_ERROR);
     });
-    return Promise.all(promises);
-  })
-  .then((responses) => {
-    context.fileIds = responses.reduce((hash, elem) => {
-      if (elem) hash[elem.file] = elem.id;
-      return hash;
-    }, {});
-    return responses;
-  })
-}
+  }
 
-function gasCreateFile(file, type) {
-  const typeId = type === 'gs' ? 0 : 2;
-  const payload = `7|1|7|${getBaseUrl()}\|${context.gasToken}|_|makeNewFile|19|h|${file}|1|2|3|4|2|5|6|7|6|${typeId}|`;
-  return new Promise((resolve, reject) => {
-    $.ajax({
-      url: context.gasUrl,
-      headers: context.gasHeaders,
-      method: 'POST',
-      crossDomain: true,
-      data: payload,
-      dataType: 'text'
+  /*
+  * get project context with google rpc
+  * this is very volatile since it is just inferred from code
+  */
+  getGasContext() {
+    return new Promise((resolve, reject) => {
+      chrome.storage.local.get(['requestUrl' ,'requestHeaders', 'requestBody', 'gasToken'], resolve);
     })
-    .then(resolve)
-    .fail((err) => {reject(new Error('Create file failed'))})
-  })
-  .then((response) => {
-    if (!response.startsWith('//OK')) reject(new Error(`Create file '${file}.${type}' failed`));
-    const responseData = eval(response.slice(4)).filter((e) => {
-      return typeof(e) === 'object';
-    })[0];
-    const id = responseData.filter((data) => { return /^\w{8}-\w{4}-\w{4}-\w{4}-\w{12}$/.test(data) });
-    if (id.length > 0) {
-      context.fileIds[file] = id[0];
-      return id[0];
-    } else {
-      throw new Error('can not parse response');
-    }
-  });
-}
+    .then((param) => {
+      context.gasUrl = param.requestUrl;
+      context.gasHeaders = param.requestHeaders;
+      context.gasToken = param.gasToken;
+      return param.requestBody;
+    });
+  }
 
-function gasUpdateFile(file, code) {
-  const escapedCode = code.replace(/\\/g, "\\\\").replace(/\|/g, '\\!');
-  const payload = `7|1|8|${getBaseUrl()}\|${context.gasToken}|_|updateFile|19|${file}||${escapedCode}|1|2|3|4|3|5|5|5|6|7|8|`;
-  let headers = context.gasHeaders;
-  Object.assign(headers, { 'file-id': context.fileIds[file]});
-  return new Promise((resolve, reject) => {
-    $.ajax({
-      url: context.gasUrl,
-      headers: headers,
-      method: 'POST',
-      crossDomain: true,
-      data: payload,
-      dataType: 'text'
+  getGasCode() {
+    const files = $('.item').toArray().map((e) => {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/9b7bf2b597e67365f1e9002c784237b58657e4cc/src/gas-api.js#L76"></span>‘files’ is assigned a value but never used.</td>
    </tr>
  </tbody>
</table>